### ISSUE TITLE
query contributors for repos in ecosystem-dashboard?

### ISSUE THREAD

**Issue by SgtPooki** [ipfs-shipyard/ecosystem-dashboard#387]
as requested, posting our slack convo as issue:

currently, for [PL-compass](https://github.com/SgtPooki/PL-Compass), i’m getting top contributors for a repo by querying github at [https://api.github.com/repos/${repoName}/contributors?q=contributions&order=desc](https://api.github.com/repos/$%7BrepoName%7D/contributors?q=contributions&order=desc) is there a similar query I can make to ecosystem-research to get this data?

The only way I could figure doing that from ecosystem-research data is that I would need to:

1. query repos
2. query contributors
3. cross-reference the two

> So you could pull all the events for a given repository and then group events by users (actor column) to get unique contributors
> But all events will include things like users who have made stars and forks events on that repo, so you might need to exclude certain event types
> We could also make a repo contributor end point that does the grouping directly in the database, wouldn't take very long
> @andrew

**Comment by andrew** [comment]
![Screenshot 2022-03-16 at 12 37 41](https://user-images.githubusercontent.com/1060/158591511-2a7012fa-3b5e-4118-b531-764bfd7b9343.png)

https://ipfs.ecosystem-dashboard.com/repositories/web3-storage/web3.storage/contributors gives a paginated html list and https://ipfs.ecosystem-dashboard.com/repositories/web3-storage/web3.storage/contributors.json returns all contributors in one big json array.

The counts may not be quite the same as from the github contributors api as the github one only counts commits, where as I'm counting any kind of contribution to the repo (apart from starring and forking) and I'm also excluding bots.

**Comment by SgtPooki** [comment]
That's excellent @andrew! Thanks! which table has the contributors' relational status (i.e. core vs. collab vs. other) and is it possible to get that information in the API as well?

**Comment by andrew** [comment]
Yeah we can add that in, might need to paginate the api response to keep it speedy

**Comment by andrew** [comment]
Added `core` field in https://github.com/ipfs-shipyard/ecosystem-dashboard/commit/051fcc431dea03057852856c8324f079abe1dd5d
